addons/web_editor/static/src/components/upload_progress_toast/upload_service.js

```diff
@@ -67,6 +67,13 @@ export const uploadService = {
              * @param {Function} onUploaded
              */
             uploadFiles: async (files, {resModel, resId, isImage}, onUploaded) => {
+                const extensionRegex = /(?:\.([^.]+))?$/;
+                const match = extensionRegex.exec(files[0].name);
+                await rpc('/web_editor/get_valid_file', {
```

Comment:
Yes, i also think about that but i did't get any `webkitRelativePath` as you can see in image below 
![image](https://github.com/odoo/odoo/assets/98956978/137138ba-fda1-4848-a07d-99f77798c868)

Reply:
No, it would not solve my issue because although the solution enables the selection of multiple files, choosing a folder still triggers a traceback error (`error : UncaughtPromiseError`) in the `master`.

Reply:
I don't really understand what is causing the issue and I can not reproduce it as my operating system does not allow me to pick a directory (I use Linux Mint and Firefox/Chrome). The error you got does not give me any hint about what went wrong.

If you select a directory, the browser should pass a file list and our code should upload the files one by one. There is maybe another issue on `master`. I will investigate a bit :smiley:

For 16.0, can I open a new PR with the code I suggested? I think it's a cool improvement.